src/common/url-parser.ts

```diff
@@ -33,4 +48,26 @@ export class UrlParser {
     private areUrlProtocolsEqual(urlAObj: URL, urlBObj: URL): boolean {
         return urlAObj.protocol === urlBObj.protocol;
     }
+
+    private tryParseUrl(url?: string): URL | null {
```

Comment:
Not sure if I understand why this parameter or the one below are optional? If you're worried about them being null/undefined, wouldn't it make sense to change the parameter type as such (assuming that's the case, I'd imagine we'd want to change the types on the public methods as well?).